nd2reader/parser.py

```diff
@@ -247,6 +247,28 @@ def _channel_offset(self):
         """
         return {channel: n for n, channel in enumerate(self.metadata["channels"])}
 
+    def _check_unwanted_bytes(self, image_group_data, image_data_start, height, width):
+        number_of_true_channels = int(len(image_group_data[4:]) / (height * width))
+        n_unwanted_bytes = (len(image_group_data[image_data_start:]))%(height*width)
+        if not n_unwanted_bytes:
+            return False
+        assert 0 == n_unwanted_bytes % height, "Unexpected unwanted bytes distribution."
+        byte_ids = range(image_data_start+height*number_of_true_channels, len(image_group_data)-n_unwanted_bytes+1, height*number_of_true_channels)
+        all_zero_bytes = all([0 == image_group_data[byte_ids[i]+i] for i in range(len(byte_ids))])
+        if not all_zero_bytes:
+            warnings.warn(f'Identified {n_unwanted_bytes} unwanted non-zero bytes in the ND2 file.')
```

Comment:
So if I understand correctly, also in this case the image is not parsed? What would then be the logic for issuing a warning instead of raising an exception?
In any case, the message could be a bit more clear, something like "{n_unwanted_bytes} unexpected non-zero bytes were found in the ND2 file, the file could not be parsed."

Reply:
The reasoning here was to just issue a warning and let the parser crash down the line raising the appropriate exception at the crashing point. I could raise an exception here instead, what do you think?

I agree on clarifying the message!